src/crypto/tls/ech.go

```diff
@@ -723,6 +739,14 @@ func echDecodeClientHelloInner(encodedData, outerData, outerSessionId []byte) []
 		panic(err) // Host encountered internal error
 	}
 
+	// Read the padding.
+	for !s.Empty() {
+		var zero uint8
+		if !s.ReadUint8(&zero) || zero != 0 {
```

Comment:
It might be good to change the order here to `!s.ReadUint8(&zero) || zero != 0`. I think golang executes first the left-hand and then the right-hand, so maybe you want first to check the latter rather than the first as that check might be faster...

Reply:
The first expression (`!s.ReadUint8(&zero)`) needs to be evaluated first because it sets `zero` as a side-effect.

Reply:
ah.. interesiting.